common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -420,18 +432,88 @@ private void notifyListeners() {
         }
         EventExecutor executor = executor();
         if (executor.inEventLoop()) {
```

Comment:
Why don't you do the stack protection here? If the stack depth is too great, schedule it on the executor. Is there concern that the executor may run the OneTimeTask immediately (not enqueuing it)?